staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -457,3 +459,58 @@ func ParseApplySetParentRef(parentRefStr string, mapper meta.RESTMapper) (*Apply
 	}
 	return &ApplySetParentRef{Name: name, RESTMapping: mapping}, nil
 }
+
+// Prune deletes any objects from the apiserver that are no longer in the applyset.
+func (a *ApplySet) Prune(ctx context.Context, o *ApplyOptions) error {
+	printer, err := o.ToPrinter("pruned")
+	if err != nil {
+		return err
+	}
+	opt := &ApplySetDeleteOptions{
+		CascadingStrategy: o.DeleteOptions.CascadingStrategy,
+		DryRunStrategy:    o.DryRunStrategy,
+		GracePeriod:       o.DeleteOptions.GracePeriod,
+
+		Printer: printer,
+
+		IOStreams: o.IOStreams,
+	}
+
+	if err := a.pruneAll(ctx, o.DynamicClient, o.VisitedUids, opt); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FindObjectsToPrune returns the list of objects that will be pruned.
+// Calling this instead of Prune can be useful for dry-run / diff behaviour.
+func (a *ApplySet) FindObjectsToPrune(ctx context.Context, dynamicClient dynamic.Interface, visitedUids sets.Set[types.UID]) ([]PruneObject, error) {
```

Comment:
Why have `FindObjectsToPrune` that is public wrapping the very similarly named `findAllObjectsToPrune` that is private?

Reply:
Good point, removed the extra layer.